### PR TITLE
[WIP] WAT 379 code editor

### PR DESCRIPTION
**Pull request by luckyRoGGeR**
Code editor with syntax highlighting and autocomplete.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "@types/chai": "4.1.4",
     "@types/mocha": "5.2.4",
     "@types/node": "10.5.2",
+    "@types/webpack": "4.4.5",
```
Comment (johnthecat) [inline package.json]: move types into recorder-fronend

File: packages/testring-recorder-frontend/package.json [hunk 2]
```diff
@@ -20,5 +21,12 @@
     "ts-loader": "^4.4.2",
     "webpack": "4.15.1",
     "webpack-cli": "3.0.8"
+  },
+  "dependencies": {
+    "css-loader": "1.0.0",
+    "style-loader": "^0.21.0",
+    "monaco-editor-webpack-plugin": "^1.4.0",
+    "react-ace": "6.1.4",
```
Comment (johnthecat) [inline packages/testring-recorder-frontend/package.json]: remove react-ace from deps

File: packages/testring-recorder-frontend/package.json [hunk 3]
```diff
@@ -20,5 +21,12 @@
     "ts-loader": "^4.4.2",
     "webpack": "4.15.1",
     "webpack-cli": "3.0.8"
+  },
+  "dependencies": {
```
Comment (johnthecat) [inline packages/testring-recorder-frontend/package.json]: move all `dependencies` into `devDependencies`

File: packages/testring-recorder-frontend/src/index.tsx [hunk 4]
```diff
@@ -1,8 +1,9 @@
+import * as React from 'react';
```
Comment (dividedby0) [inline packages/testring-recorder-frontend/src/index.tsx]: In this package TS runs with flag allowSyntheticDefaultImports.
@johnthecat can't we import it just like `import React from 'react';`

**Pull request merged**